src/pebble.py

```diff
@@ -89,14 +89,12 @@ def stop_agent(self, container: ops.Container) -> None:
 
         Args:
             container: The agent workload container.
-
-        Raises:
-            APIError: if something went wrong with pebble requesting service stop.
         """
         try:
-            container.stop(self.state.jenkins_agent_service_name)
-        except ops.pebble.APIError as exc:
-            if f'service "{self.state.jenkins_agent_service_name}" does not exist' in exc.message:
-                return
-            raise
+            # use get_service to check if service should be stopped rather than stopping and
+            # catching ops.pebble.APIError and parsing error message to determine type of error.
+            container.get_service(self.state.jenkins_agent_service_name)
+        except ops.ModelError:
```

Comment:
What's this exception?

Reply:
This exception is raised when `container.get_service` is called and the service doesn't exist. It is a workaround for having to parse pebble's APIError :)

Reply:
I see. Thx